tests/test_plugins.py

```diff
@@ -0,0 +1,115 @@
+
+    def tearDown(self):
+        self.page.delete()
+        self.home.delete()
+        self.superuser.delete()
+
+    def test_player_plugin(self):
+        plugin = add_plugin(
+            placeholder=self.placeholder,
+            plugin_type=VideoPlayerPlugin.__name__,
+            language=self.language,
+        )
+        self.assertEqual(plugin.plugin_type, "VideoPlayerPlugin")
+
+    def test_source_plugin(self):
+        plugin = add_plugin(
+            placeholder=self.placeholder,
+            plugin_type=VideoSourcePlugin.__name__,
+            language=self.language,
+        )
+        self.assertEqual(plugin.plugin_type, "VideoSourcePlugin")
+
+    def test_track_plugin(self):
+        plugin = add_plugin(
+            placeholder=self.placeholder,
+            plugin_type=VideoTrackPlugin.__name__,
+            language=self.language,
+        )
+        self.assertEqual(plugin.plugin_type, "VideoTrackPlugin")
+
+    def test_plugin_structure(self):
+        video_file = get_filer_file("test_file.mp4")
+        track_file = get_filer_file("test_track.vtt")
+        request_url = self.page.get_absolute_url(self.language) + "?toolbar_off=true"
+
+        parent = add_plugin(
+            placeholder=self.placeholder,
+            plugin_type=VideoPlayerPlugin.__name__,
+            language=self.language,
+            template="default",
+        )
+        self.page.publish(self.language)
+        self.assertEqual(parent.get_plugin_class_instance().name, "Video player")
+
+        with self.login_user_context(self.superuser):
+            response = self.client.get(request_url)
+
+        self.assertIn(b"Your browser doesn't support this video format.", response.content)
+
+        child = add_plugin(
+            target=parent,
+            placeholder=self.placeholder,
+            plugin_type=VideoSourcePlugin.__name__,
+            language=self.language,
+            source_file=video_file,
+        )
+        self.page.publish(self.language)
+        self.assertEqual(child.source_file.label, "test_file.mp4")
+
+        with self.login_user_context(self.superuser):
+            response = self.client.get(request_url)
+
+        self.assertIn(b"<video controls", response.content)
+        self.assertContains(response, video_file.label)
+
+        track = add_plugin(
+            target=parent,
+            placeholder=self.placeholder,
+            plugin_type=VideoTrackPlugin.__name__,
+            language=self.language,
+            kind="subtitles",
+            src=track_file,
+            srclang=self.language,
+        )
+        self.page.publish(self.language)
+        self.assertEqual(track.src.label, "test_track.vtt")
+
+        with self.login_user_context(self.superuser):
+            response = self.client.get(request_url)
+
+        self.assertIn(b"<track kind", response.content)
+        self.assertContains(response, track_file.label)
+
+        # cleanup
+        video_file.delete()
```

Comment:
the execution might not reach this point when there is an error above and the files would not get cleaned up. 

can this be moved to the teardown?